app.js

```diff
@@ -1,36 +1,109 @@
-var tar = require('tar-fs');
-var fs = require('fs');
+const tar = require('tar-fs');
+const fs = require('fs');
 
-const PORT = 5000;
+const PORT = 5016;
 
 // Two main eventEmitters of the application
 const server = require('./server');
 const javaBox = require('./javaBox');
 
-// Test
-server.on('testDocker', (socket) => {
-    javaBox.emit('testDocker', socket);
-}).on('testJava', (socket) => {
-    javaBox.emit('testJava', socket);
-}).on('runJava', runSingleClass);
 
+server.on('runJava', runSingleClass);
+javaBox.on('result', giveFeedBack);
 
-function runSingleClass(socket, data) {
 
-    console.log(data);
+function runSingleClass(clientId, main, files) {
 
-    let dirPath = './dockerFiles/singleClass';
-    let tarPath = dirPath + '.tar';
-    let filePath = dirPath + '/Main.java';
+    const dirPath = `./dockerFiles/${clientId}`;
+    const tarPath = `${dirPath}.tar`;
 
-    fs.mkdir(dirPath, (err) => console.log('directory created'));
-    fs.createWriteStream(filePath).write(data);
+    const makeTarAndRun = () => {
 
-    tar.pack(dirPath).pipe(fs.createWriteStream(tarPath));
+        tar.pack(dirPath).pipe(fs.createWriteStream(tarPath));
+        javaBox.emit('runJava', clientId, main, tarPath);
+    };
 
-    javaBox.emit('runJava', socket, tarPath);
+    const writeFiles = () => {
```

Comment:
fs.createWriteStream(filePath).write(file.data, fileAdded); at line 59 and fs.unlink(`${dirPath}/${oldFile}`, fileDeleted); at line 68 have fileAdded and fileDeleted as callbacks, these two decrease filesToDelete or filesToAdd int variables respectively and both call tryTarAndRun, that checks whether filesToDelete and filesToAdd are both 0 and, if they are, call makeTarAndRun